commit d4af904d455c8348d128568f87beb4f74099578e
Author: Dawit Alemayehu <adawit@kde.org>
Date:   Sat Jul 7 15:08:20 2012 -0400

    Do not use the user's default browser configuration to open XML files.
    
    BUG: 291177
    FIXED-IN: 4.9.0
    REVIEW: 105476

diff --git a/kio/kio/krun.cpp b/kio/kio/krun.cpp
index 53ab878..785966c 100644
--- a/kio/kio/krun.cpp
+++ b/kio/kio/krun.cpp
@@ -1177,9 +1177,9 @@ void KRun::init()
         KMimeType::Ptr mime = KMimeType::findByUrl(d->m_strURL, d->m_mode, true /*local*/);
         assert(mime);
         kDebug(7010) << "MIME TYPE is " << mime->name();
-        if (!d->m_externalBrowser.isEmpty() && (
-               mime->is(QLatin1String("text/html")) ||
-               mime->is(QLatin1String("application/xml")))) {
+        if (!d->m_externalBrowser.isEmpty() &&
+            (mime->is(QLatin1String("text/html")) ||
+             mime->is(QLatin1String("application/xhtml+xml")))) {
             if (d->runExecutable(d->m_externalBrowser)) {
                 return;
             }
